src/service.ts

```diff
@@ -331,6 +332,42 @@ export class Service {
     }
     return uri;
   }
+
+  static downloadProject(project: Project, uri?: string) {
+    const zipFile: JSZip = new JSZip();
+    let zipName: string = "download.zip";
+    if (!isUndefined(uri)) {
+      zipName = `${uri}.zip`;
+    }
+    function defineZip(file: File, folder?: any) {
```

Comment:
The name `defineZip` is not very clear. This function is adding zip files recursively to the archive. Perhaps we can call it `addZipFile`, or `addZipFileRecursively`, or `zip`, etc.